### DISCUSSION TITLE
output http module: buffer variable does not take effect

### DISCUSSION THREAD

**Discussion by qinjunhang** [fluent/fluentd#4111] [Q&A]
![image](https://user-images.githubusercontent.com/55073294/227476203-b5e2f209-9f4c-490f-9235-261596728fd4.png)

# td-agent --version
td-agent 4.4.2 fluentd 1.15.3 (e89092ce1132a933c12bb23fe8c9323c07ca81f5)

Hello, when I use the file module, the <buffer name> can get the variable, but I can’t get it in the http module, please help me

**Comment by daipom** [comment]
I don't know, but could it be due to the use of single quotations?

**Reply by qinjunhang** [reply]
I experimented and it has nothing to do with single quotes

**Reply by daipom** [reply]
Sorry, certainly it has nothing to do with that.

And I can confirm `in_http` can get the buffer key variable as `out_file`.
It is a common feature, so there is no difference between plugins.

If you can't, old buffer files may remain, or records that don't have the `name` field may flow in.
